### PR TITLE
[RAPTOR-3866] Catch exception from user's code, log it and return proper response

### PR DESCRIPTION
**Pull request by zohar-mizrahi**
Two major changes:
* Catch user's exceptions that might occur during predictions, log them and return a proper response
* Pump up mlpiper version dependency to mlpiper v2.4.0

**Note:** the tests for this PR will not pass until the `mlpiper v2.4.0` will be pushed to DataRobot artifactory

### CODE REVIEW

**Review by alexromanyuk — APPROVED** [review]
:+1:

**Comment by yakov-g** [comment]
jarvis test cmrunner_integration

**Review by rvorobii — APPROVED** [review]

**Comment by yakov-g** [comment]
@rvorobii 
COuld you take a look?
With these changes  `test_e2e_predict_fails` fails: with error:
```
>           assert response.status_code == 500  # error occurs
E           assert 513 == 500
E            +  where 513 = <Response [513]>.status_code
```

Do we want to update tests?

**Comment by rvorobii** [comment]
> @rvorobii
> COuld you take a look?
> With these changes `test_e2e_predict_fails` fails: with error:
> 
> ```
> >           assert response.status_code == 500  # error occurs
> E           assert 513 == 500
> E            +  where 513 = <Response [513]>.status_code
> ```
> 
> Do we want to update tests?

The tests is correct.
513 should be only returned if exception happened before prediction server started.
513 return drum initilization errors, but if prediction server is already running - initilization was fine, and there should be no 513.

File: custom_model_runner/datarobot_drum/resource/components/Python/uwsgi_component/uwsgi_serving.py [hunk 1]
```diff
@@ -145,6 +145,10 @@ def predict(self, url_params, form_params):
                 # this counter is managed by uwsgi
                 self._total_predict_requests.increase()
                 self._predict_calls_count += 1
+        except Exception as ex:
+            self._logger.error(ex)
+            response_status = HTTP_513_DRUM_PIPELINE_ERROR
```
Comment (rvorobii) [inline custom_model_runner/datarobot_drum/resource/components/Python/uwsgi_component/uwsgi_serving.py:150]: @yakov-g raised a good point - why 513 and not 500?
513 is designed only for errors happened during drum initilization.
prediction errors should be 500

**Review by rvorobii — CHANGES_REQUESTED** [review]
looks like there should be an error with 500, not with 513

File: custom_model_runner/datarobot_drum/resource/components/Python/uwsgi_component/uwsgi_serving.py (hunk 1 above)
Comment (rvorobii) [reply custom_model_runner/datarobot_drum/resource/components/Python/uwsgi_component/uwsgi_serving.py:150]: also, looks like this ticket is about this issue
could you please have a look at close it if it's fixed https://datarobot.atlassian.net/browse/RAPTOR-3751?

Comment (yakov-g) [reply custom_model_runner/datarobot_drum/resource/components/Python/uwsgi_component/uwsgi_serving.py:150]: thanks
I'll take a look, if @zohar-mizrahi won't

**Review by rvorobii — APPROVED** [review]

**Comment by datarobot-jarvis** [comment]
:dollar: Total cost of executing tests in Jenkins: $**5.87** USD.

[What is this all about?](http://bit.ly/2BahK4V)

**Pull request merged**